### PR TITLE
chore: add comments

### PR DESCRIPTION
**Pull request by pscott**
Adding documentation comments on the most important function

### CODE REVIEW

File: starknet/src/authenticators/eth_tx.cairo [hunk 1]
```diff
@@ -3,6 +3,9 @@ use sx::types::{Strategy, IndexedStrategy, Choice};
 
 #[starknet::interface]
 trait IEthTxAuthenticator<TContractState> {
```
Comment (Orland0x) [inline starknet/src/authenticators/eth_tx.cairo:5]: any reason these comments are different format to others?

File: starknet/src/authenticators/stark_sig.cairo [hunk 2]
```diff
@@ -1,8 +1,21 @@
 use starknet::ContractAddress;
 use sx::types::{Strategy, IndexedStrategy, Choice};
 
+/// See https://community.starknet.io/t/snip-off-chain-signatures-a-la-eip712/98029
 #[starknet::interface]
 trait IStarkSigAuthenticator<TContractState> {
+    /// Authenticates a propose transaction using the starknet EIP71-equivalent signature scheme.
```
Comment (Orland0x) [inline starknet/src/authenticators/stark_sig.cairo]: EIP712, similarly for funcs below

File: starknet/src/execution_strategies/eth_relayer.cairo [hunk 3]
```diff
@@ -11,6 +11,17 @@ mod EthRelayerExecutionStrategy {
     #[storage]
     struct Storage {}
 
+    /// Forwards the execution to the L1<->Starknet bridge to the contract located in the `payload` argument.
```
Comment (Orland0x) [inline starknet/src/execution_strategies/eth_relayer.cairo]: Forwards the proposal data to the L1 execution strategy specified in the `payload` argument via the Starknet<->L1 bridge.

File: starknet/src/execution_strategies/eth_relayer.cairo [hunk 4]
```diff
@@ -11,6 +11,17 @@ mod EthRelayerExecutionStrategy {
...
+    /// Since this contract does not check who is calling it, it is the responsibility of the L1 contract to
+    /// check that the caller is indeed an authorized space contract (this information is sent to the bridge).
+    ///
+    /// # Arguments
+    ///
+    /// * proposal - The proposal to execute.
+    /// * votes_for - The number of votes for the proposal.
+    /// * votes_against - The number of votes against the proposal.
+    /// * votes_abstain - The number of votes abstaining from the proposal.
+    /// * payload - Should be a tuple of the form `(l1_execution_strategy, l1_execution_hash)`.
```
Comment (Orland0x) [inline starknet/src/execution_strategies/eth_relayer.cairo]: payload - An array containing the serialized L1 execution strategy address and the L1 execution hash.

File: starknet/src/execution_strategies/vanilla.cairo [hunk 5]
```diff
@@ -19,6 +19,8 @@ mod VanillaExecutionStrategy {
         }
     }
 
+    /// The vanilla execution strategy is a dummy execution strategy that simply increments a `_num_executed` variable for every
+    /// new execute proposal. It uses the `SimpleQuorum` method to determine whether a proposal is accepted or not.
```
Comment (Orland0x) [inline starknet/src/execution_strategies/vanilla.cairo]: newly

File: starknet/src/interfaces/i_quorum.cairo [hunk 6]
```diff
@@ -1,4 +1,6 @@
+/// Option trait that execution strategies can decide to implement.
```
Comment (Orland0x) [inline starknet/src/interfaces/i_quorum.cairo]: Optional

File: starknet/src/interfaces/i_voting_strategy.cairo [hunk 7]
```diff
@@ -3,6 +3,18 @@ use sx::types::UserAddress;
 
 #[starknet::interface]
 trait IVotingStrategy<TContractState> {
+    /// 
+    ///
+    /// # Arguments
+    ///
+    /// * `timestamp` - The timestamp to use to compute the voting power.
```
Comment (Orland0x) [inline starknet/src/interfaces/i_voting_strategy.cairo]: voting power at

File: starknet/src/space/space.cairo [hunk 8]
```diff
@@ -5,28 +5,71 @@ use sx::types::{
...
 #[starknet::interface]
 trait ISpace<TContractState> {
-    // State 
+    // -- View functions --
+
+    /// Returns the owner of the contract.
     fn owner(self: @TContractState) -> ContractAddress;
-    fn max_voting_duration(self: @TContractState) -> u32;
+
+    /// The voting delay to wait for between a proposal is created and the start of the voting period.
```
Comment (Orland0x) [inline starknet/src/space/space.cairo]: The voting delay in seconds between when a proposal is created and the start of the voting period.

File: starknet/src/space/space.cairo [hunk 9]
```diff
@@ -5,28 +5,71 @@ use sx::types::{
...
+    fn voting_delay(self: @TContractState) -> u32;
+
+    /// Returns the minimum voting duration. Once this duration has elapsed,
+    /// a proposal can be finalized. Most DAOs should set `min_voting_duration` and
```
Comment (Orland0x) [inline starknet/src/space/space.cairo]: technically a proposal can be executed before min voting duration has elapsed as we check that in the execution strategy. think just a simple comment like in sx-evm is fine: https://github.com/snapshot-labs/sx-evm/blob/df37a70673067300a3a6c0541740117a86d0bc59/src/interfaces/space/ISpaceState.sol#L13C6-L13C6

File: starknet/src/space/space.cairo [hunk 10]
```diff
@@ -5,28 +5,71 @@ use sx::types::{
...
+    /// `max_voting_duration` to the same value.
     fn min_voting_duration(self: @TContractState) -> u32;
+
+    /// Returns the maximum voting duration. Once this duration has elasped, a proposal
+    /// can be finalized. No other votes can come in.
```
Comment (Orland0x) [inline starknet/src/space/space.cairo]: Once this duration has elapsed, no additional votes can be cast.

File: starknet/src/space/space.cairo [hunk 11]
```diff
@@ -5,28 +5,71 @@ use sx::types::{
...
+
+    /// An array of allowed voting strategies. Voting strategies are used to determine the voting power of a voter.
     fn voting_strategies(self: @TContractState, index: u8) -> Strategy;
+
+    /// The current active voting strategies. Each bit represents whether the proposal is active or not.
+    /// These are stored at the proposal level when a proposal is created, meaning when a new voting strategy is added or removed
+    /// it only affects subsequent proposals, not ongoing ones.
     fn active_voting_strategies(self: @TContractState) -> u256;
+
+    /// The next voting strategy index. Starts at 0.
```
Comment (Orland0x) [inline starknet/src/space/space.cairo]: The index of the next available voting strategy. Starts at 0

File: starknet/src/space/space.cairo [hunk 12]
```diff
@@ -41,13 +84,35 @@ trait ISpace<TContractState> {
         metadata_uri: Array<felt252>,
         dao_uri: Array<felt252>,
     );
+    /// Creates a new proposal. Users must go through an authenticator before calling this function.
+    ///
+    /// # Arguments
+    ///
+    /// * `author` - The author of the proposal.
+    /// * `metadata_uri` - The metadata URI of the proposal. Could be a link to a discussion page, etc.
+    /// * `execution_strategy` - The execution strategy of the proposal. This is the contract that will be called
```
Comment (Orland0x) [inline starknet/src/space/space.cairo]: The execution strategy for the proposal, consisting of a strategy address and an execution payload.

File: starknet/src/space/space.cairo [hunk 13]
```diff
@@ -41,13 +84,35 @@ trait ISpace<TContractState> {
...
+    /// Casts a vote on a given proposal. Users must go through an authenticator before calling this function.
+    ///
+    /// # Arguments
+    ///
+    /// * `voter` - The voter.
+    /// * `proposal_id` - The proposal id to vote on.
+    /// * `choice` - The choice to vote for.
+    /// * `user_voting_strategies` - The voting strategies that the user wants to vote with. This allows to only 
+    ///                             query the `user_voting_strategies` for voting power instead of iterating through all
+    ///                             allowde voting strategies.
```
Comment (Orland0x) [inline starknet/src/space/space.cairo]: The strategies to use to compute the voter's voting power, each consisting of a strategy index and an array of user provided parameters.

File: starknet/src/types/indexed_strategy.cairo [hunk 14]
```diff
@@ -1,8 +1,12 @@
 use sx::utils::math;
 
+/// An indexed strategy is used to access a strategy in an array of
```
Comment (Orland0x) [inline starknet/src/types/indexed_strategy.cairo]: A strategy identified by a unique index.

File: starknet/src/types/proposal.cairo [hunk 15]
```diff
@@ -16,15 +16,26 @@ const TWO_POWER_32: u128 = 0x100000000;
 const TWO_POWER_64: u128 = 0x10000000000000000;
 const TWO_POWER_96: u128 = 0x1000000000000000000000000;
 
+/// Definition of a proposal.
```
Comment (Orland0x) [inline starknet/src/types/proposal.cairo]: Data stored when a proposal is created. The proposal state at any time consists of this struct along with the corresponding for, against, and abstain vote counts. The proposal status is a function of the proposal state as defined in the execution strategy for the proposal.

File: starknet/src/types/proposal.cairo [hunk 16]
```diff
@@ -16,15 +16,26 @@ const TWO_POWER_32: u128 = 0x100000000;
...
     max_end_timestamp: u32,
+    /// The finalization status of the proposal.
     finalization_status: FinalizationStatus,
+    /// The hash of the execution payload.
     execution_payload_hash: felt252,
+    /// The address of the execution strategy.
     execution_strategy: ContractAddress,
+    /// The address of the proposal author.
     author: UserAddress,
+    /// A snapshot of the active voting strategies at the time of proposal creation.
```
Comment (Orland0x) [inline starknet/src/types/proposal.cairo]: Bit array where the index of each each bit corresponds to whether the voting strategy at that index is active at the time of proposal creation.

File: starknet/src/types/strategy.cairo [hunk 17]
```diff
@@ -3,7 +3,9 @@ use starknet::{StorageBaseAddress, Store, SyscallResult};
 
 #[derive(Clone, Drop, Option, Serde, starknet::Store)]
 struct Strategy {
```
Comment (Orland0x) [inline starknet/src/types/strategy.cairo:6]: A strategy identified by an address

File: starknet/src/types/update_settings_calldata.cairo [hunk 18]
```diff
@@ -17,6 +20,7 @@ struct UpdateSettingsCalldata {
     voting_strategies_to_remove: Array<u8>,
 }
 
+// TODO: use `Default` trait
```
Comment (Orland0x) [inline starknet/src/types/update_settings_calldata.cairo:23]: can this be done yet?

File: starknet/src/authenticators/stark_sig.cairo (hunk 2 above)
Comment (pscott) [reply starknet/src/authenticators/stark_sig.cairo]: copilot rekted me

File: starknet/src/authenticators/eth_tx.cairo (hunk 1 above)
Comment (pscott) [reply starknet/src/authenticators/eth_tx.cairo:5]: lazyness :D  015174552e7f75c6882639b367d1cd54b8d6fb1b

File: starknet/src/authenticators/stark_sig.cairo (hunk 2 above)
Comment (pscott) [reply starknet/src/authenticators/stark_sig.cairo]: 56f3bd50f8916897f0cc3f36063235fb9571d1bf

File: starknet/src/execution_strategies/eth_relayer.cairo (hunk 3 above)
Comment (pscott) [reply starknet/src/execution_strategies/eth_relayer.cairo]: 85ee323b4503b8cafc928cd2372ae7fd504fad38

File: starknet/src/execution_strategies/eth_relayer.cairo (hunk 4 above)
Comment (pscott) [reply starknet/src/execution_strategies/eth_relayer.cairo]: b738b3a56ccf77db864cc0eed9775465171f196d

File: starknet/src/execution_strategies/vanilla.cairo (hunk 5 above)
Comment (pscott) [reply starknet/src/execution_strategies/vanilla.cairo]: 21516ccf83626b87757fef3df075c585a0930c71

File: starknet/src/interfaces/i_quorum.cairo (hunk 6 above)
Comment (pscott) [reply starknet/src/interfaces/i_quorum.cairo]: c9524b1366754e6da7a42dcea7aafdb9f9ae7023

File: starknet/src/interfaces/i_voting_strategy.cairo (hunk 7 above)
Comment (pscott) [reply starknet/src/interfaces/i_voting_strategy.cairo]: 1763db1b0f83dc4c00d5bd59cf2958508c427358

File: starknet/src/space/space.cairo (hunk 8 above)
Comment (pscott) [reply starknet/src/space/space.cairo:13]: 3b561eb2d3c5ee7376822f546c0fe081368ddb63

File: starknet/src/space/space.cairo (hunk 9 above)
Comment (pscott) [reply starknet/src/space/space.cairo]: 6f348d79f7d366db2741e52a881e17708515063c

File: starknet/src/space/space.cairo (hunk 8 above)
Comment (pscott) [reply starknet/src/space/space.cairo]: 6f348d79f7d366db2741e52a881e17708515063c

Comment (pscott) [reply starknet/src/space/space.cairo]: 3b561eb2d3c5ee7376822f546c0fe081368ddb63

File: starknet/src/space/space.cairo (hunk 10 above)
Comment (pscott) [reply starknet/src/space/space.cairo]: 3b561eb2d3c5ee7376822f546c0fe081368ddb63

File: starknet/src/space/space.cairo (hunk 11 above)
Comment (pscott) [reply starknet/src/space/space.cairo]: 6f348d79f7d366db2741e52a881e17708515063c

File: starknet/src/space/space.cairo (hunk 12 above)
Comment (pscott) [reply starknet/src/space/space.cairo]: 90a90ccc7f75db0456bd19d7cace99868b8f3f57

File: starknet/src/space/space.cairo (hunk 13 above)
Comment (pscott) [reply starknet/src/space/space.cairo]: 6d158bf4a956c397eaea2bd5c89c2b3d44f824e7

File: starknet/src/space/space.cairo (hunk 12 above)
Comment (pscott) [reply starknet/src/space/space.cairo]: 7240ee167dd3f87aa02debe96a521af6b0b6cfd1

File: starknet/src/space/space.cairo (hunk 11 above)
Comment (pscott) [reply starknet/src/space/space.cairo]: 97aa327876b9eaa8c501dbb0b9381b6b813065f8

File: starknet/src/space/space.cairo (hunk 10 above)
Comment (pscott) [reply starknet/src/space/space.cairo]: 90a90ccc7f75db0456bd19d7cace99868b8f3f57

File: starknet/src/types/indexed_strategy.cairo (hunk 14 above)
Comment (pscott) [reply starknet/src/types/indexed_strategy.cairo]: 19b443bbef885a0c30318a95c568e12f42da3f06

File: starknet/src/types/proposal.cairo (hunk 15 above)
Comment (pscott) [reply starknet/src/types/proposal.cairo]: e436b9d57c73975e3bece13603cc9ef4d5deeae0

File: starknet/src/types/proposal.cairo (hunk 16 above)
Comment (pscott) [reply starknet/src/types/proposal.cairo]: 495aa612cd6309f4c716b43928fe3e934eac70b0

File: starknet/src/types/strategy.cairo (hunk 17 above)
Comment (pscott) [reply starknet/src/types/strategy.cairo:6]: ab1bd7e96e89859a0feba050045e65338ab9b9e6

File: starknet/src/types/update_settings_calldata.cairo (hunk 18 above)
Comment (pscott) [reply starknet/src/types/update_settings_calldata.cairo:23]: I tried to but wasn't working. Will try in another PR

**Review by Orland0x — APPROVED** [review]
uTACK

**Pull request merged**